llvm/lib/CodeGen/SelectionDAG/SelectionDAGISel.cpp

```diff
@@ -180,6 +182,15 @@ static const bool ViewDAGCombine1 = false, ViewLegalizeTypesDAGs = false,
                   ViewSchedDAGs = false, ViewSUnitDAGs = false;
 #endif
 
+#define ISEL_DUMP(F, X)                                                        \
+  do {                                                                         \
+    if (llvm::DebugFlag && (isCurrentDebugType(DEBUG_TYPE) ||                  \
+                            (isCurrentDebugType(ISEL_DUMP_DEBUG_TYPE) &&       \
+                             isFunctionInPrintList(F)))) {                     \
```

Comment:
This seems a bit heavy to check at every site, right? I think we could at least cache `isFunctionInPrintList(F)` at the start of the pass.  Off the top of my head, I can't think of a nicer way to check the `DEBUG_TYPE` in the current implementation, but maybe alternate structure would help?

Maybe I'm overthinking this detail though, and perf isn't so important if you're debug printing something ...

Reply:
Done. Now it stores the filter result in a class member variable of `SelectionDAGISel`. Interesting side note: `isCurrentDebugType`, which is used by `LLVM_DUMP`, actually does a linear search to filter debug types every time it's called. But I guess the size of `-debug-only` is not big so it's fine for majority of the time.